src/main/java/com/movie/store/controllers/UserController.java

```diff
@@ -24,8 +24,8 @@ public UserController(UserService userService,
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto getUserByEmail(@RequestParam String email) {
-        User user = userService.findByEmail(email).get();
+    public UserResponseDto getUserByEmail(Authentication authentication) {
```

Comment:
you don't need to change this method, it should return user by given email is you send HTTP request